### DISCUSSION TITLE
Documentation source links for packages in monorepos

### DISCUSSION THREAD

**Discussion by richard-viney** [gleam-lang/gleam#3760] [Ideas & suggestions]
Generated source links in the docs currently expect the Gleam package being published to be at the root of the `repository` specified in `gleam.toml`. At the moment there doesn't appear to be any other config:

<img width="249" alt="image" src="https://github.com/user-attachments/assets/2ee446e7-a1ba-4ea7-9180-a9bdc519bd66">

Could we extend this with some kind of `path` or `prefix` option to allow for publication of Gleam packages stored in monorepos, i.e. ones that don't reside at the root of their repository?

**Comment by lpil** [comment]
Good idea!

**Comment by richard-viney** [comment]
Implemented in https://github.com/gleam-lang/gleam/pull/3765.
